### PR TITLE
test for various endpoints in the external_result module

### LINKED ISSUE #2033: Adds test for various endpoints in the external_result module
part of epic issue - #1372

### PR DESCRIPTION
**Pull request by praffq**
## Proposed Changes

- Added test for various endpoints in the external_result module

### Associated Issue

- fixes #2033 

## Merge Checklist

- [x] Tests added/fixed
- [x] Linting Complete

_*Only PR's with test cases included and passing lint and test pipelines will be reviewed*_

@coronasafe/care-backend-maintainers @coronasafe/care-backend-admins

### CODE REVIEW

File: care/facility/tests/test_external_result_api.py [hunk 1]
```diff
@@ -12,6 +13,69 @@ def setUpTestData(cls) -> None:
...
+
+    def test_list_external_result(self):
+        response = self.client.get("/api/v1/external_result/")
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_retrieve_external_result(self):
+        response = self.client.get(
+            f"/api/v1/external_result/{self.external_result.id}/"
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
```
Comment (sainak) [inline care/facility/tests/test_external_result_api.py]: assert the response as well

Comment (praffq) [reply care/facility/tests/test_external_result_api.py]: what should I test in `test_list_external_result` , i added a test for checking count, what else?

Comment (sainak) [reply care/facility/tests/test_external_result_api.py]: you can add more external results objects and use filters defined [here](https://github.com/coronasafe/care/blob/db6f739a32310a6cc80e36d8c8395fdf01937aa0/care/facility/api/viewsets/patient_external_test.py#L53) to see if the filters are returning expected objects

File: care/facility/tests/test_external_result_api.py [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+                    "source": "Secondary contact aparna",
+                    "sample_collection_date": "2020-10-14",
+                    "result_date": "2020-10-14",
+                    "test_type": "Antigen",
+                    "lab_name": "Karothukuzhi Laboratory",
+                    "sample_type": "Ag-SD_Biosensor_Standard_Q_COVID-19_Ag_detection_kit",
+                    "patient_status": "Asymptomatic",
+                    "is_repeat": "NO",
+                    "patient_category": "Cat 17: All individuals who wish to get themselves tested",
+                    "result": "Negative",
```
Comment (sainak) [inline care/facility/tests/test_external_result_api.py]: now that you have a utility function you can replace this json with get_patient_external_test_data(...).update(...)

**Comment by praffq** [comment]
@sainak  Updated. Please have a look

**Review by sainak — APPROVED** [review]

**Review by vigneshhari — APPROVED** [review]

**Pull request merged**